.appveyor.yml

```diff
@@ -46,6 +46,7 @@ build_script:
       C:\%CYGWIN_DIR%\bin\bash -lc "curl -Lo /usr/local/bin/mill %MILL_URL%" &&
       C:\%CYGWIN_DIR%\bin\bash -lc 'sed -i '"'"'0,/-cp "\$0"/{s/-cp "\$0"/-cp `cygpath -w "\$0"`/}; 0,/-cp "\$0"/{s/-cp "\$0"/-cp `cygpath -w "\$0"`/}'"'"' /usr/local/bin/mill' &&
       C:\%CYGWIN_DIR%\bin\bash -lc 'chmod +x /usr/local/bin/mill' &&
-      C:\%CYGWIN_DIR%\bin\bash -lc "cd /cygdrive/c/mill && mill -i all main.test scalajslib.test")
+      C:\%CYGWIN_DIR%\bin\bash -lc "cd /cygdrive/c/mill && mill -i all __.publishLocal release" &&
```

Comment:
the cygwin job was not bootstrapping before testing, leading the tests to fail

Reply:
Can you explain why this is necessary? I don't see any reason the changes I've reviewed would cause bootstrapping issues

Reply:
Sure. I momentarily [switched output stream](https://ci.appveyor.com/project/lihaoyi/mill/build/1.0.397/job/rcg2s135uvlj9ia6) to System.out in failing test to understand what was happening. 

As far as I understand, without bootstrapping [the foreign0 param](https://github.com/lihaoyi/mill/pull/291/files#diff-34a1cc8c4134f41e770d9a1012a485aaR11) was not defined, and the [wrapping code using it] would not compile (https://github.com/lihaoyi/mill/pull/291/files#diff-c38a2bb70bc7af1e007f759ca8a015aeR143)

Reply:
Running the unit tests via `mill all main.test` should use the freshly-compiled version of Mill; the fact that it's including `foreign0` in the code wrapper is evidence of this. Thus I do not see why that freshly-compiled Mill would not be able to handle `foreign0` during compilation.

Furthermore, the Travis build ran the foreign module tests also without bootstrapping, but successfully.

Can you dig into this a bit deeper? We should figure out what is going on here rather than just working around the issue.

Reply:
Will do